noobaa/noobaa-core
# Rework S3 object version ID and marker check

### Explain the changes
1. Move the version ID check to happen before `read_object_md` is called, which takes care of both namespacestore and backingstore queries - while also potentially eliminating the case in which the store would enter a Rejected state
2. Remove the max-line linter error in `test_s3_ops`
3. Check for empty strings in all API calls containing a version ID
4. Check for empty strings in all API calls containing a version ID marker

### Issues: Fixed #xxx / Gap #xxx
1. [2265562](https://bugzilla.redhat.com/show_bug.cgi?id=2265562)

### Testing Instructions:
1. Upload an object to a bucket
5. Use the AWS CLI (`s3api` with `--version-id ""`) to try and access an existing object's version ID by providing an empty string
6. Verify that NooBaa throws the appropriate error (`An error occurred (InvalidArgument) when calling the GetObject operation: Version id cannot be the empty string`)
